docs/introduction-to-smart-contracts.rst

```diff
@@ -57,6 +57,10 @@ and overwrite your number, but the number will still be stored in the history
 of the blockchain. Later, we will see how you can impose access restrictions
 so that only you can alter the number.
 
+.. note::
+    All identifiers (contract names, function names and variable names) are restricted to
+    the ASCII character set. It is possible to store UTF-8 encoded data in string variables.
```

Comment:
We should add a note somewhere stating that `UTF-8` encoded strings might look identical in an editor but can still be different at the byte level.

Reply:
Added as a warning block below.